src/casts/specify-representation.rkt

```diff
@@ -122,20 +122,22 @@ but a static single assignment is implicitly maintained.
      (begin$
        (assign$ val e1)
        (assign$ type (sr-expr e2))
-       (assign$ tag (sr-get-tag type TYPE-TAG-MASK))
-       (Switch
-           type
-         (let* ([shifted-imm (op$ %<< val DYN-IMDT-SHIFT)]
-                [tag-shifted-imm
-                 (lambda ([tag : D0-Expr])
-                   : D0-Expr
-                   (op$ binary-or shifted-imm tag))])
-           `([(,data:TYPE-INT-RT-VALUE)  . ,(tag-shifted-imm DYN-INT-TAG)]
-             [(,data:TYPE-BOOL-RT-VALUE) . ,(tag-shifted-imm DYN-BOOL-TAG)]
-             [(,data:TYPE-UNIT-RT-VALUE) . ,(tag-shifted-imm DYN-UNIT-TAG)]
-             [(,data:TYPE-CHAR-RT-VALUE) . ,(tag-shifted-imm DYN-CHAR-TAG)]))
-         ;; Notice that float types fall into this case also
-         (sr-alloc "dynamic_boxed" DYN-BOXED-TAG `(("" . ,val) ("" . ,type)))))]))
+       (let* ([shifted-imm (op$ %<< val DYN-IMDT-SHIFT)]
+              [tag-shifted-imm
+               (lambda ([tag : D0-Expr])
+                 : D0-Expr
+                 (op$ binary-or shifted-imm tag))])
+         (case$ type
+           [(data:TYPE-INT-RT-VALUE) (tag-shifted-imm DYN-INT-TAG)]
+           [(data:TYPE-BOOL-RT-VALUE) (tag-shifted-imm DYN-BOOL-TAG)]
+           [(data:TYPE-UNIT-RT-VALUE) (tag-shifted-imm DYN-UNIT-TAG)]
+           [(data:TYPE-CHAR-RT-VALUE) (tag-shifted-imm DYN-CHAR-TAG)]
+           [(data:TYPE-FLOAT-RT-VALUE)
+            (sr-alloc
+             "dynamic_boxed" DYN-BOXED-TAG `(("" . ,val) ("" . ,type)))]
```

Comment:
I don't now that this matters but this change add an additional branch to actually generated code.

Reply:
Done.